lib/mongo/retryable.rb

```diff
@@ -403,5 +407,10 @@ def log_retry(e, options = nil)
       end
       Logger.logger.warn "#{message} due to: #{e.class.name} #{e.message}"
     end
+
+    # Retry writes on MMAPv1 should raise an actionable error, so we change the error message.
+    def raise_unsupported_error(e)
+      raise e, "This MongoDB deployment does not support retryable writes. Please add retryWrites=false to your connection string."
```

Comment:
The original exception class and message need to be included in the final message following the `#{e.class}: #{e}` pattern.

Reply:
Since the class of exception is not changing, it may also make sense to preserve the backtrace which looks like this:
<pre>
new_e = Error::OperationFailure.new(new_msg)
new_e.set_backtrace(e.backtrace)
raise new_e
</pre>

Reply:
Is there another place is the driver where we compose multiple error messages like above? I would have assumed that the helpful error message here ("This MongoDB deployment does not support retryable writes...") should come first, but your comment about not including a trailing period is making me think it should come after the original exception class and message.

Reply:
The pattern I usually use is when translating one exception class to another, the message of the new class goes first, but when adding information to the existing exception and/or message, the information gets added at the end.

Example exceptions which get extended error messages are:

- NoServerAvailable (https://jira.mongodb.org/browse/RUBY-1787)
- SocketError/SocketTimeoutError (https://github.com/mongodb/mongo-ruby-driver/pull/988/files, https://jira.mongodb.org/browse/RUBY-1391)
- Mongo::Auth::Unauthorized

Reply:
Thank you!